From f8741fe95adb425f7fdaba5860671fc096384c39 Mon Sep 17 00:00:00 2001
From: Stephen Warren <swarren@nvidia.com>
Date: Thu, 31 Jan 2019 17:14:17 -0700
Subject: [PATCH] Fix operation on NVIDIA Jetson
X-NVConfidentiality: public

Fix a couple of issues that showed up on NVIDIA Jetson running Linux for
Tegra (L4T). I'm not sure if these issues show up due to the kernel
version (4.9), the 64-bit ARM architecture, or some other reason.

- Specify the protocol for the socket. Without this, a non-TCP socket is
  created, and TCP connection attempts never reach it.

- Specify a backlog of 1. Without this, the kernel rejects all new TCP
  connections.
---
 linux/src/xvcd.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/linux/src/xvcd.c b/linux/src/xvcd.c
index f6e71d568603..af0a381924c3 100644
--- a/linux/src/xvcd.c
+++ b/linux/src/xvcd.c
@@ -271,7 +271,7 @@ int main(int argc, char **argv)
 	// Listen on port 2542.
 	//
 	
-	s = socket(AF_INET, SOCK_STREAM, 0);
+	s = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
 	
 	if (s < 0)
 	{
@@ -292,7 +292,7 @@ int main(int argc, char **argv)
 		return 1;
 	}
 	
-	if (listen(s, 0) < 0)
+	if (listen(s, 1) < 0)
 	{
 		perror("listen");
 		return 1;
-- 
2.20.1

